launchable/test_runners/nunit.py

```diff
@@ -15,9 +16,26 @@ def build_path(e: Element):
         pp = e.parent.tags.get('path') or []    # type: ignore
     if e.name == "test-suite":
         # <test-suite>s form a nested tree structure so capture those in path
-        e.tags['path'] = pp + [{'type': e.attrs['type'], 'name': e.attrs['name']}]
+        pp = pp + [{'type': e.attrs['type'], 'name': e.attrs['name']}]
     if e.name == "test-case":
-        e.tags['path'] = pp + [{'type': 'TestCase', 'name': e.attrs['name']}]
+        pp = pp + [{'type': 'TestCase', 'name': e.attrs['name']}]
+
+    if len(pp) > 0:
+        def split_filepath(path: str) -> List[str]:
+            # Supports Linux and Windows
+            if '/' in path:
+                return path.split('/')
+            else:
+                return path.split('\\')
+
+        # "Assembly" type containts full path at a cutomer's environment
+        # remove file path prefix in Assembly
+        e.tags['path'] = [
+            {**path, 'name': split_filepath(path['name'])[-1]}
```

Comment:
TIL. I guess `path['name']` in `**path` gets overridden by the `'name': exp`

Looks like `{'foo':3, 'foo':5}` evaluates to `{'foo':5}` which makes sense but somehow I thought that would result in an error.

Reply:
It's a common pattern in lightweight languages. For example, JavaScript has `{...obj}`, Ruby has `{**obj}`.